src/client/components/TablePaginated/Filters/Country/Country.tsx

```diff
@@ -40,8 +40,11 @@ const Country: React.FC<Props> = (props: Props) => {
   return (
```

Comment:
@minotogna Currently in the TablePaginated we handle the redux state with the api `path`. 
<img width="348" alt="image" src="https://github.com/user-attachments/assets/1a9fe272-0ff4-496e-99b2-76742984c614">


When we switch cycles that `path` stays the same, so we use the same state in Redux for the same table but in different cycle. 
With that in mind, I noticed this when testing for this PR :

https://github.com/user-attachments/assets/a792f4c1-8437-476b-bc93-c5de02a311fd

Since the state to manage the table is the same, the filter values are kept. 

To fix it I already tried something like:
```
export const useResetFilter = (props: Props) => {
  const { fieldName, path } = props
  const dispatch = useAppDispatch()
  const { assessmentName, cycleName } = useCycleRouteParams()

  return useEffect(() => {
    dispatch(TablePaginatedActions.resetFilter({ fieldName, path }))
  }, [assessmentName, cycleName, dispatch, fieldName, path])
}
```
But that wouldn't really work as the filters would be reset also when the hook is mounted (e.g when navigating countries -> invitations -> countries).

Should we instead consider the assessment/cycle when we store the TablePaginated state?

Let me know what you think

Reply:
@minotogna @sorja Added `useResetOnUnmount` to TablePaginated. Let me know what you guys think